pkg/checks/podRestarts/podRestarts.go

```diff
@@ -159,46 +154,59 @@ func (prc *Checker) doChecks() error {
-			if observation.Time.Before(time.Now().Add(-time.Hour)) {
-				prc.RestartObservations[podName] = deleteFromSlice(i, prc.RestartObservations[podName])
+			if observation.Time.After(time.Now().Add(-time.Hour)) {
+				s = append(s, observation)
 			}
-			// Find the most recent PRC observation
+		}
+		prc.RestartObservations[podName] = s
+	}
+	for podName, restartObservations := range prc.RestartObservations {
+		var mostRecentObservation RestartCountObservation
+		var secondMostRecentObservation RestartCountObservation
+		// Find the most recent PRC observation
+		for _, observation := range restartObservations {
 			if observation.Time.After(mostRecentObservation.Time) {
 				mostRecentObservation = observation
 			}
-			// Find the second most recent PRC observation
-			if !observation.Time.IsZero() && observation.Time != mostRecentObservation.Time && observation.Time.After(secondMostRecentObservation.Time) {
+
+		}
+		// Find the second most recent PRC observation
+		for _, observation := range restartObservations {
+			if observation.Time.Before(mostRecentObservation.Time) && observation.Time.After(secondMostRecentObservation.Time) {
 				secondMostRecentObservation = observation
 			}
 		}
-	}
-	// If the most recent observation restart count is less than the previous observation restart count, delete all the
-	// entries that arent the most recent one, assuming that the pod was restarted but still has the same name.  This is a new
-	// pod with a reset restart counter but has the same exact name as the previous pod that existed but was deleted
- 	// we see this in stateful sets
-	if mostRecentObservation.Count < secondMostRecentObservation.Count {
-		for podName, restartObservations := range prc.RestartObservations {
-			for i, observation := range restartObservations{
-				if observation != mostRecentObservation {
-					// delete all the old entries
-					prc.RestartObservations[podName] = deleteFromSlice(i, prc.RestartObservations[podName])
-				}
-			}
+		fmt.Println(podName, "most recent", mostRecentObservation)
+		fmt.Println(podName, "second most recent", secondMostRecentObservation)
+		// If the most recent observation restart count is less than the previous observation restart count, delete all the
+		// entries that arent the most recent one, assuming that the pod was restarted but still has the same name.  This is a new
+		// pod with a reset restart counter but has the same exact name as the previous pod that existed but was deleted
+		// we see this in stateful sets
+		if mostRecentObservation.Count < secondMostRecentObservation.Count {
+			// delete all the observations
+			fmt.Println(podName, "was reset to the most recent observation because it has been recreated.")
+			delete(prc.RestartObservations, podName)
+			//Add back in the most recent observation
+			prc.RestartObservations[podName] = append(prc.RestartObservations[podName], mostRecentObservation)
```

Comment:
because you just deleted this key in the map, you can just set the results, without an append

Reply:
you could actually not even do a delete, just overwrite with the single entry

Reply:
I'm deleting the map entry entirely which is a `map[string][]RestartCountObservation` and adding back in a single, non sliced `RestartCountObservation`.  

I could just clear the existing slice (rather than deleting the entire map entry) and append the new `RestartCountObservation`, but to simply overwrite I would have to change the non sliced `RestartCountObservation` into a slice for the storage of only one expected value above.

Either way I'm doing an append and it felt cleaner to just delete the entire map key and replace it since this is a brand new pod.